src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentQueryExtensions.cs

```diff
@@ -0,0 +1,18 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace OrchardCore.ContentManagement
+{
+    public static class ContentQueryExtensions
+    {
+        public static async Task<ContentItem> FirstOrDefaultAsync(this IQuery<ContentItem> query, IContentManager contentManager)
+            => await contentManager.LoadAsync(await query.FirstOrDefaultAsync());
+
+        public static async Task<IEnumerable<ContentItem>> ListAsync(this IQuery<ContentItem> query, IContentManager contentManager)
+            => await contentManager.LoadAsync(await query.ListAsync());
+
+        public static IAsyncEnumerable<ContentItem> ToAsyncEnumerable(this IQuery<ContentItem> query, IContentManager contentManager)
```

Comment:
I'd add some docs on these as well as `ContentQueryOfTIndexExtensions`.

Reply:
Okay, can you point me where it could be adde in the docs?

Reply:
I mean code comments on the methods.

Reply:
Oups, okay no problem

Reply:
Comments that you will have to review ;)

Reply:
Oh no :(. Don't overdo it, just a sentence on what each of these are here for and why would you use them.

Reply:
Done